src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -378,7 +378,62 @@ def deleteBlocks(self):
           If no => do nothing (check again next cycle)
         """
         # FIXME: figure out the proper logic for rule block deletion
+        # TODO: Update metadata after deleting block rules
         logging.info("Starting deleteBlocks methods --> IMPLEMENT-ME!!!")
+        logging.info("Starting deleteBlocks methods --> I'M ON IT...")
+
+        #Get list of blocks that can be deleted
+        blockDict = self.findDeletableBlocks.execute(transaction=False)
+
+        if not blockDict:
+            logging.info("No blocks were found")
+            return
+
+        deletableBlocks = []
+        containerDict = {}
+		#Populate containerDict, assigning each block to its correspondant container
+        for blockName in blockDict:
+            containerList = self.rucio.listParentDIDs(blockName, scope=self.scope)
+            for cont in containerList:
+                if cont['name'] not in containerDict:
+                    containerDict[cont['name']] = {'blocks':[], 'rse':[], 'copies':0}
+                containerDict[cont['name']]['blocks'].append(blockName)
+
+		
+        for contName in containerDict:
+            cont = containerDict[contName]
+            rules = self.rucio.listDataRules(contName, self.scope)
+            #Retrieve requested copies and RSEs
+            for rule in rules:
+                if rule['account'] == self.rucioAcct:
+                    cont['copies'] += rule['copies']
+                    cont['rse'].extend(self.rucio.evaluateRSEExpression(rule['rse_expression'], useCache=True, returnTape=True))
+
+            #Get RSE in which the block is available
+            availableRSEs = self.rucio.getReplicaInfoForBlocks(block=cont['blocks'])
+            
+            for rses in availableRSEs:
+                #Count how many of the copies were requested by this account
+                actualCopies = len(set(rses['replica']) & cont['rse'])
```

Comment:
This should be `set(rses['replica']) & set(cont['rse']))`. I fixed it during local pylint tests but must have reversed the changes by accident.